expression/expression.go

```diff
@@ -780,7 +787,40 @@ func ColumnInfos2ColumnsAndNames(ctx sessionctx.Context, dbName, tblName model.C
 		}
 		columns = append(columns, newCol)
 	}
-	return columns, names
+	// Resolve virtual generated column.
+	mockSchema := NewSchema(columns...)
+	save := ctx.GetSessionVars().StmtCtx.IgnoreTruncate
+	defer func() {
+		ctx.GetSessionVars().StmtCtx.IgnoreTruncate = save
+	}()
+	ctx.GetSessionVars().StmtCtx.IgnoreTruncate = true
```

Comment:
Is it thread-safe?

Reply:
It's used in the planner and it's used serail.